File: web/src/api/checkins/attendance.js
import service from '@/utils/request'

export const createAttendance = (data) => {
  return service({
    url: '/attendance/createAttendance',
    method: 'post',
    data
  })
}


export const cloneAttendance = (data) => {
  return service({
    url: '/attendance/cloneAttendance',
    method: 'post',
    data
  })
}

export const createAttendanceArea = (data) => {
  return service({
    url: '/attendance/createAttendanceArea',
    method: 'post',
    data
  })
}

export const deleteAttendance = (params) => {
  return service({
    url: '/attendance/deleteAttendance',
    method: 'delete',
    params
  })
}

export const deleteAttendanceArea = (params) => {
  return service({
    url: '/attendance/deleteAttendanceArea',
    method: 'delete',
    params
  })
}

export const deleteAttendanceByIds = (params) => {
  return service({
    url: '/attendance/deleteAttendanceByIds',
    method: 'delete',
    params
  })
}

export const updateAttendance = (data) => {
  return service({
    url: '/attendance/updateAttendance',
    method: 'put',
    data
  })
}

export const findAttendance = (params) => {
  return service({
    url: '/attendance/findAttendance',
    method: 'get',
    params
  })
}

export const findAttendanceArea = (params) => {
  return service({
    url: '/attendance/findAttendanceArea',
    method: 'get',
    params
  })
}

export const getAttendanceList = (params) => {
  return service({
    url: '/attendance/getAttendanceList',
    method: 'get',
    params
  })
}

export const getAttendancePublic = (params) => {
  return service({
    url: '/attendance/getAttendancePublic',
    method: 'get',
    params
  })
}

export const statsByAgencyCategory = (data) => {
  return service({
    url: '/attendance/statsByAgencyCategory',
    method: 'post',
    data
  })
}


export const statsScatterPlot = (data) => {
  return service({
    url: '/attendance/statsScatterPlot',
    method: 'post',
    data
  })
}

export const statsTrendLine = (data) => {
  return service({
    url: '/attendance/statsTrendLine',
    method: 'post',
    data
  })
}

File: server/router/checkins/attendance.go
package checkins

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type AttendanceRouter struct{}

func (s *AttendanceRouter) InitAttendanceRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	attendanceRouter := Router.Group("attendance").Use(middleware.OperationRecord())
	attendanceRouterWithoutRecord := Router.Group("attendance")
	attendanceRouterWithoutAuth := PublicRouter.Group("attendance")
	{
		attendanceRouter.POST("createAttendance", attendanceApi.CreateAttendance)
		attendanceRouter.POST("createAttendanceArea", attendanceApi.CreateAttendanceArea)
		attendanceRouter.POST("cloneAttendance", attendanceApi.CloneAttendance)
		attendanceRouter.POST("statsByAgencyCategory", attendanceApi.StatsByAgencyCategory)
		attendanceRouter.POST("statsScatterPlot", attendanceApi.StatsScatterPlot)
		attendanceRouter.POST("statsTrendLine", attendanceApi.StatsTrendLine)

		attendanceRouter.DELETE("deleteAttendance", attendanceApi.DeleteAttendance)
		attendanceRouter.DELETE("deleteAttendanceArea", attendanceApi.DeleteAttendanceArea)
		attendanceRouter.DELETE("deleteAttendanceByIds", attendanceApi.DeleteAttendanceByIds)
		attendanceRouter.PUT("updateAttendance", attendanceApi.UpdateAttendance)

	}
	{
		attendanceRouterWithoutRecord.GET("findAttendance", attendanceApi.FindAttendance)
		attendanceRouterWithoutRecord.GET("findAttendanceArea", attendanceApi.FindAttendanceArea)
		attendanceRouterWithoutRecord.GET("getAttendanceList", attendanceApi.GetAttendanceList)
	}
	{
		attendanceRouterWithoutAuth.GET("getAttendancePublic", attendanceApi.GetPublicAttendanceList)
	}
}
